crates/top/rerun/src/commands/entrypoint.rs

```diff
@@ -231,6 +231,28 @@ If no arguments are given, a server will be hosted which a Rerun SDK can connect
     #[clap(long)]
     renderer: Option<String>,
 
+    /// Overwrites hardware acceleration option for video decoding.
+    ///
+    /// By default uses the last provided setting, which is `auto` if never configured.
+    ///
+    /// Depending on the decoder backend, these settings are merely hints and may be ignored.
+    /// However, they can be useful in some situations to work around issues.
+    ///
+    /// Possible values:
+    ///
+    /// * `auto`
+    /// May use hardware acceleration if available and compatible with the codec.
+    ///
+    /// * `prefer_software`
+    /// Should use a software decoder even if hardware acceleration is available.
+    /// If no software decoder is present, this may cause decoding to fail.
+    ///
+    /// * `prefer_hardware`
+    /// Should use a hardware decoder.
+    /// If no hardware decoder is present, this may cause decoding to fail.
+    #[clap(long, verbatim_doc_comment)]
+    video_decoder: Option<String>,
```

Comment:
You can use an `enum` here instead

```rs
#[derive(ArgEnum, Clone, Copy, Debug, PartialEq, Eq)]
#[clap(rename_all = "snake_case")]
enum VideoDecoder {
    /// May use hardware acceleration if available and compatible with the codec.
    Auto,

    /// …
    PreferHardware,

    /// …
    PreferSoftware,
}
```

```suggestion
    video_decoder: Option<VideoDecoder>,
```

Reply:
Hmm I'll have to look up how that picks up on the flexible parsing then

Reply:
looks like for that I have to infect re_renderer with clap. I'd rather not have that dependency if it can be avoided

Reply:
oh and much worse still: we don't even have re_renderer as a dependency here necessarily - this crate may or may not know about the native viewer directly!

Reply:
👍